xmtp_mls/src/groups/device_sync/consent_sync.rs

```diff
@@ -0,0 +1,174 @@
+use super::*;
+use crate::{
+    storage::key_value_store::{KVStore, Key},
+    Client, XmtpApi,
+};
+use xmtp_id::scw_verifier::SmartContractSignatureVerifier;
+
+impl<ApiClient, V> Client<ApiClient, V>
+where
+    ApiClient: XmtpApi + Clone,
+    V: SmartContractSignatureVerifier + Clone,
+{
+    pub async fn send_consent_sync_request(&self) -> Result<(String, String), DeviceSyncError> {
+        let request = DeviceSyncRequest::new(DeviceSyncKind::Consent);
+        self.send_sync_request(request).await
+    }
+
+    pub async fn reply_to_consent_sync_request(
+        &self,
+        pin_code: &str,
+    ) -> Result<DeviceSyncReplyProto, DeviceSyncError> {
+        let (_msg, request) = self.pending_sync_request(DeviceSyncKind::Consent).await?;
+
+        self.verify_pin(&request.request_id, pin_code)?;
+
+        let consent_records = self.syncable_consent_records()?;
+
+        let reply = self
+            .send_syncables(&request.request_id, &[consent_records])
+            .await?;
+
+        Ok(reply)
+    }
+
+    pub async fn process_consent_sync_reply(&self) -> Result<(), DeviceSyncError> {
+        let conn = self.store().conn()?;
+
+        // load the request_id
+        let request_id: Option<String> =
+            KVStore::get(&conn, &Key::ConsentSyncRequestId).map_err(DeviceSyncError::Storage)?;
+        let Some(request_id) = request_id else {
+            return Err(DeviceSyncError::NoReplyToProcess);
+        };
+
+        // process the reply
+        self.process_sync_reply(&request_id).await
+    }
+
+    fn syncable_consent_records(&self) -> Result<Vec<Syncable>, DeviceSyncError> {
```

Comment:
Is there some document I can look at to see how we are expecting to merge these consent records?

I'm thinking about situations like:

Client A blocks Address A
Client B allows Address A
Client C gets created

(and a bunch of variations on that idea)

How do we make sure that all clients get a consistent view of who is allowed and blocked?